doc/source/operations/upgrading.rst

```diff
@@ -0,0 +1,919 @@
+
+Source the ``kayobe-env`` script:
+
+.. code-block:: console
+
+   source kayobe-env [--environment <env>]
+
+Export the Ansible Vault password:
+
+.. code-block:: console
+
+   export KAYOBE_VAULT_PASSWORD=$(cat /path/to/vault/password/file)
+
+Next we must upgrade the Ansible control host.  Tasks performed here include:
+
+- Install updated Ansible collection and role dependencies from Ansible Galaxy.
+- Generate an SSH key if necessary and add it to the current user's authorised
+  keys.
+- Upgrade Kolla Ansible locally to the configured version.
+
+To upgrade the Ansible control host:
+
+.. code-block:: console
+
+   kayobe control host upgrade
+
+Syncing Release Train artifacts
+-------------------------------
+
+New `StackHPC Release Train <../configuration/release-train>` content should be
+synced to the local Pulp server. This includes host packages (Deb/RPM) and
+container images.
+
+.. _sync-rt-package-repos:
+
+To sync host packages:
+
+.. code-block:: console
+
+   kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-sync.yml
+   kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-publish.yml
+
+Once the host package content has been tested in a test/staging environment, it
+may be promoted to production:
+
+.. code-block:: console
+
+   kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-promote-production.yml
+
+To sync container images:
+
+.. code-block:: console
+
+   kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-container-sync.yml
+   kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-container-publish.yml
+
+Build locally customised container images
+-----------------------------------------
+
+.. note::
+
+   The container images are provided by StackHPC Release Train are
+   suitable for most deployments. In this case, this step can be skipped.
+
+In some cases it is necessary to build some or all images locally to apply
+customisations. In order to do this it is necessary to set
+``stackhpc_pulp_sync_for_local_container_build`` to ``true`` before
+:ref:`syncing container images <sync-rt-package-repos>`.
+
+To build the overcloud images locally and push them to the local Pulp server:
+
+.. code-block:: console
+
+   kayobe overcloud container image build --push
+
+It is possible to build a specific set of images by supplying one or more
+image name regular expressions:
+
+.. code-block:: console
+
+   kayobe overcloud container image build --push ironic- nova-api
+
+Pull container images to hosts
+------------------------------
+
+Pulling container images from the local Pulp server to the control plane hosts
+can take a considerable time, because images are only synced from Ark to the
+local Pulp on demand, and there is potentially a large fan-out. Pulling images
+in advance of the upgrade moves this step out of the maintenance/change window.
+
+.. code-block:: console
+
+   kayobe overcloud container image pull
+
+Preview overcloud service configuration changes
```

Comment:
I tried this config diff process during the Zed upgrade on SMS and something went a bit wrong. I'm not sure if it was an issue with my config or the commands & I don't remember exactly what i used to correct it. I'll try it again for Antelope and make a separate PR with changes if needed. I'm happy with this as-is for now, just wanted to note my experience